client/my-sites/navigation/navigation.jsx

```diff
@@ -2,46 +2,60 @@
  * External dependencies
  */
 import React from 'react';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import observe from 'lib/mixins/data-observe';
 import SitePicker from 'my-sites/picker';
 import Sidebar from 'my-sites/sidebar';
+import { getCurrentUser } from 'state/current-user/selectors';
+import { getCurrentLayoutFocus } from 'state/ui/layout-focus/selectors';
+import { setNextLayoutFocus, setLayoutFocus } from 'state/ui/layout-focus/actions';
 
-module.exports = React.createClass( {
+const MySitesNavigation = React.createClass( {
 	displayName: 'MySitesNavigation',
 
-	mixins: [ observe( 'sites', 'user', 'layoutFocus' ) ],
+	mixins: [ observe( 'sites' ) ],
 
 	preventPickerDefault( event ) {
 		event.preventDefault();
 		event.stopPropagation();
 	},
 
 	render() {
-		var layoutFocus = this.props.layoutFocus;
-
 		return (
 			<div className="sites-navigation">
 				<SitePicker
-					layoutFocus={ layoutFocus }
+					setNextLayoutFocus={ this.props.setNextLayoutFocus }
```

Comment:
Again, ideally these components (`<SidePicker />` and `<Sidebar />`) can `connect` themselves to state.